docs/app/styles/chart/barchart/barchart.component.ts

```diff
@@ -0,0 +1,163 @@
+import { Component, OnInit, ViewChild, ElementRef, Input, ViewEncapsulation, HostListener } from '@angular/core';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'hc-demo-barchart',
+  template: `<div class="d3-bar-chart" #chart></div>`,
+  styleUrls: ['./barchart.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class BarchartComponent implements OnInit {
+    @ViewChild('chart') private chartContainer: ElementRef;
+    @Input() private data: Array<any>;
+    private margin: any = { top: 20, bottom: 45, left: 30, right: 20};
+    private chart: any;
```

Comment:
If this was in a library component that was reused, I would recommend strongly typing these `any`s but as these charts are just examples I think it's okay.